chainntnfs/btcdnotify/btcd.go

```diff
@@ -447,6 +449,25 @@ func (b *BtcdNotifier) RegisterSpendNtfn(outpoint *wire.OutPoint) (*chainntnfs.S
 
 	b.notificationRegistry <- ntfn
 
+	// The following conditional checks to ensure that when a spend notification
+	// is registered, the output hasn't already been spent. If the output
+	// is no longer in the UTXO set, the chain will be rescanned from the point
+	// where the output was added. The rescan will dispatch the notification.
+	if txout, err := b.chainConn.GetTxOut(&outpoint.Hash, outpoint.Index, true); err != nil {
+		return nil, err
+	} else if txout == nil {
+		if transaction, err := b.chainConn.GetRawTransactionVerbose(&outpoint.Hash); err != nil {
```

Comment:
This block here seems to have some unnecessary nesting which I think can be eliminated. The block can be reformatted to look like the following: 
```
txout, err := b.chainConn.GetTxOut(&outpoint.Hash, outpoint.Index, true)
if err != nil {
    return nil, err
}

// Output has been spent.... 
if txout == nil {
    tx, err := b.chainConn.GetRawTransactionVerbose(&outpoint.Hash)
    if err != nil {
        return nil, err
    }

    blockHash, err := wire.NewShaHashFromStr(transaction.BlockHash)
    if err != nil {
        return nil, err
    }

    ops := []*wire.OutPoint{outpoint}
    if err := b.chainConn.Rescan(blockhash, nil, ops); err != nil {
        log message
        return nil, err
    } 
}
```